nasa/earthdata-search
# EDSC-3929: Fix spatial subsetting with Harmony when using MultiPolygons

# Overview

### What is the feature?

Users are encountering difficulties in spatially subsetting data when the input shapes are of MultiPolygon type. This is because EDSC splits MultiPolygons into separate Polygon features, but was omitting the `properties` key in the process. 

The `properties` key is required by the geojson specification. When it is not included, Harmony will return a `Failed to parse shapefile` error. That error is not currently displayed in EDSC, and a Harmony job is not created, so users don't know why their Harmony jobs fail in EDSC.

### What is the Solution?

The existing `properties` of the MultiPolygon are copied to the new split Polygon features. In the unlikely event there are no input properties, an empty `properties` object is created, which is compliant with the geojson spec.

### What areas of the application does this impact?

Spatial subsetting through Harmony.

# Testing

### Reproduction steps

The bug was reported against TEMPO L2 data, so that is what I've been using for testing.

- **Environment for testing:** UAT
- **Collection to test with:** C1262899916-LARC_CLOUD

1. Search for the collection and view its granules
2. Spatially subset using the attached shapefile. 
3. Pan the map view the Philadelphia, PA region
4. Click _twice_ on the orange-outlined shape
    - We are trying to select this shape
    - If instead it selects a granule, zoom back in and try again
    - <img width="447" alt="image" src="https://github.com/nasa/earthdata-search/assets/10775395/f3bbaa77-58af-4b2a-b001-72143511386f">
5. You should see the number of granules reduced from 45 to 6
6. Download some of these granules via Harmony's L2 Cloud Subsetter
7. The Harmony job should be successful and return 

Bonus: confirm the downloaded data conforms to the input shapefile's geometry in Panoply:
<img width="600" alt="image" src="https://github.com/nasa/earthdata-search/assets/10775395/336da13c-87e7-428b-ab0e-fd41dedd85bb">


### Attachments

A shapefile with a single MultiPolygon feature:
[ozone_Pa.zip](https://github.com/user-attachments/files/15821624/ozone_Pa.zip)

# Checklist
